src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs/breadcrumb.rs

```diff
@@ -0,0 +1,501 @@
+        let relative_position = default();
+
+        let shape_system = scene.shapes.shape_system(PhantomData::<background::Shape>);
+        scene.views.main.remove(&shape_system.shape_system.symbol);
+        scene.views.breadcrumbs.add(&shape_system.shape_system.symbol);
+
+        let shape_system = scene.shapes.shape_system(PhantomData::<icon::Shape>);
+        shape_system.shape_system.set_pointer_events(false);
+        scene.views.main.remove(&shape_system.shape_system.symbol);
+        scene.views.breadcrumbs.add(&shape_system.shape_system.symbol);
+
+        let shape_system = scene.shapes.shape_system(PhantomData::<separator::Shape>);
+        shape_system.shape_system.set_pointer_events(false);
+        scene.views.main.remove(&shape_system.shape_system.symbol);
+        scene.views.breadcrumbs.add(&shape_system.shape_system.symbol);
+
+        let symbol = glyph_system.sprite_system().symbol();
+        scene.views.main.remove(&symbol);
+        scene.views.breadcrumbs.add(&symbol);
+
+        Self{logger,view,icon,separator,display_object,glyph_system,label,info,animations
+            ,is_selected,relative_position}.init()
+    }
+
+    fn init(self) -> Self {
+        self.add_child(&self.view);
+        self.view.add_child(&self.separator);
+        self.separator.add_child(&self.icon);
+        self.icon.add_child(&self.label);
+
+        let color  = if self.is_selected() { FULL_COLOR } else { TRANSPARENT_COLOR };
+
+        self.label.set_font_size(TEXT_SIZE);
+        self.label.set_font_color(color);
+        self.label.set_text(&self.info.method_pointer.name);
+        self.label.set_position(Vector3(ICON_RADIUS + ICON_RIGHT_MARGIN, -TEXT_SIZE/3.0, 0.0));
+
+        let width  = self.width();
+        let height = self.height();
+        let offset = self.compute_separator_margin() + SEPARATOR_SIZE/2.0;
+
+        self.view.shape.sprite.size.set(Vector2::new(width,height));
+        self.fade_in(0.0);
+        let separator_size = (SEPARATOR_SIZE+PADDING*2.0).max(0.0);
+        let icon_size      = (ICON_SIZE+PADDING*2.0).max(0.0);
+        self.separator.shape.sprite.size.set(Vector2::new(separator_size,separator_size));
+        self.separator.set_position_x((offset-width/2.0).round());
+        self.icon.shape.sprite.size.set(Vector2::new(icon_size,icon_size));
+        let x_position = offset+PADDING+ICON_SIZE/2.0+LEFT_MARGIN+ICON_LEFT_MARGIN;
+        self.icon.set_position_x(x_position.round());
+
+        self
+    }
+
+    fn compute_separator_margin(&self) -> f32 {
+        self.label.font_size() * SEPARATOR_MARGIN / 6.0
+    }
+
+    fn label_width(&self) -> f32 {
+        self.info.method_pointer.name.len() as f32 * GLYPH_WIDTH
+    }
+
+    /// Get the width of the view.
+    pub fn width(&self) -> f32 {
+        let width = self.compute_separator_margin()*2.0+SEPARATOR_SIZE+ICON_SIZE+ICON_RIGHT_MARGIN;
+        let width = width+self.label_width()+LEFT_MARGIN+RIGHT_MARGIN+ICON_LEFT_MARGIN+PADDING*2.0;
```

Comment:
I do not understand this equation - could you use more variables explaining parts of it? I would like to visualize in my head what these variables mean in space.